### PR TITLE
Refactor and add framework for native jfr event support

### PR DESCRIPTION
**Pull request by zhengyu123**
This PR replaces [PR#3746](https://github.com/oracle/graal/pull/3746)

- Split JFR package into hosted (com.oracle.svm.hosted.jdk11.jfr) and core (com.oracle.svm.core.jdk11.jfr)
- Add package com.oracle.svm.core.jfr for native jfr event support (e.g. GC events)
- There will no-op JfrEventSupport for jdk8, but a concrete implementation for jdk11 or later
- Remove com.oracle.svm.jfr package

### CODE REVIEW

**Comment by jerboaa** [comment]
@zhengyu123 If you rebase this onto latest master, it should pass all the public gate tests.

File: substratevm/mx.substratevm/suite.py [hunk 1]
```diff
@@ -570,46 +599,6 @@
             "spotbugs": "false",
         },
 
-        "com.oracle.svm.jfr": {
```
Comment (christianhaeubl) [inline substratevm/mx.substratevm/suite.py:573]: I think it would be better to have the JFR implementation (and its dependencies) in a separate project. Pretty much all VM-internal events should use the `JfrNativeEventWriter` anyways (via some JDK-independent abstraction).

File: substratevm/mx.substratevm/suite.py [hunk 2]
```diff
@@ -1465,20 +1457,6 @@
           "testDistribution" : True,
         },
 
-        "SVM_TESTS_JDK11" : {
```
Comment (christianhaeubl) [inline substratevm/mx.substratevm/suite.py:1468]: Why did you delete this part?

**Comment by christianhaeubl** [comment]
I will merge the PR https://github.com/oracle/graal/pull/3720 first, as it is more complex than this PR. After that, I will merge this PR and as part of that process, I am also going to take care of all the conflicts that may come up. So, there is no need on your end to rebase this PR.

**Comment by jiekang** [comment]
@christianhaeubl Hi Christian! Any updates on this area?

**Comment by christianhaeubl** [comment]
@jiekang I would like to wait with this until the JDK 8 support is removed from Native Image as the JFR refactoring will be far simpler afterwards. The removal of the JDK 8 support is currently work in progress and should be finished soon.

**Comment by christianhaeubl** [comment]
I am closing this PR as I merged my PR in the meanwhile. Most JFR-related files are now part of the project `com.oracle.svm.core` and can therefore be used easily in other Native Image projects (e.g., in the GC).
